primitives/src/vid/advz.rs

```diff
@@ -96,15 +102,36 @@ where
                 payload_chunk_size, num_storage_nodes
             )));
         }
-        let (ck, vk) = P::trim_fft_size(srs, payload_chunk_size).map_err(vid)?;
+        let (ck, vk) = P::trim_fft_size(srs, payload_chunk_size - 1).map_err(vid)?;
+        let multi_open_domain =
+            P::multi_open_rou_eval_domain(payload_chunk_size - 1, num_storage_nodes)
+                .map_err(vid)?;
+        let eval_domain = Radix2EvaluationDomain::new(payload_chunk_size).ok_or_else(|| {
+            VidError::Internal(anyhow::anyhow!(
+                "fail to construct doman of size {}",
+                payload_chunk_size
+            ))
+        })?;
+
+        // TODO TEMPORARY: enforce power-of-2 chunk size
+        // Remove this restriction after we get KZG in eval form
+        // https://github.com/EspressoSystems/jellyfish/issues/339
+        if payload_chunk_size != eval_domain.size() {
```

Comment:
If `payload_chunk_size` is smaller. Could we simply padding zero here?

Reply:
Yes. The FFT pads with zero up to the next power of 2. This wouldn't be a problem except it screws up recovery. For a degree `d` polynomial I need `next_power_of_2(d)` points to interpolate, so we're essentially forced to use a power of 2 degree anyway. This PR is intended to be a temporary stop-gap, so I decided that it's best for now to simply enforce a power-of-2 degree at construction. This sanity check is merely a redundant backup check on the constructor.